src/components/ui/BaseButton.vue

```diff
@@ -0,0 +1,36 @@
+<template>
+  <button v-if="!link" :class="mode">
+    <slot></slot>
+  </button>
+  <router-link v-else :to="to" :class="mode">
+    <!-- router-link or router-view? -->
+    <slot></slot>
+  </router-link>
+</template>
+
+<script>
+export default {
+  props: {
+    link: {
+      type: Boolean,
+      reqired: false,
```

Comment:
you don't need to put required if it is set to false. And you have a typo here: reqired

Reply:
fix